pandas/core/indexes/category.py

```diff
@@ -381,14 +381,6 @@ def fillna(self, value, downcast=None):
         cat = self._data.fillna(value)
         return type(self)._simple_new(cat, name=self.name)
 
-    @cache_readonly
```

Comment:
hmm is _engine_type still needed then?

Reply:
yes, since codes could be int8, int16, int32, int64